doc/developers-guide/plugin-development/event-notifications.md

```diff
@@ -567,3 +567,85 @@ Where:
 - `plugin_name`: The short name of the plugin.
 - `plugin_path`: The full file path to the plugin executable.
 - `methods`: An array of RPC method names that the plugin registered.
+
+
+### `xpay_attempt_start` (v25.09 onward) 
+
+Emitted by `xpay` when part of a payment begins.  `payment_hash` and
+`groupid` uniquely identify this xpay invocation, and `partid` then identifies
+this particular attempt to pay part of that.
+
+`total_payment_msat` is the total amount (usually the invoice amount),
+which will be the same across all parts, adn `attempt_msat` is the
+amount being delivered to the destination by this part.
+
+Each element in `hops` shows the amount going into the node (i.e. with
+fees, `channel_in_msat`) and the amount we're telling it to send
+to the other end (`channel_out_msat`).  The `channel_out_msat` will
+be equal to the next `channel_in_msat.  The final
+`channel_out_msat` will be equal to the `attempt_msat`.
+
+The example shows a payment from this node via 1x2x3 (direction 1) to 035d2b1192dfba134e10e540875d366ebc8bc353d5aa766b80c090b39c3a5d885d, then via 2x3x4 (direction 0) to 022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59.
+
+```json
+{
+  "xpay_attempt_start": {
+    "payment_hash": "f5a6a059a25d1e329d9b094aeeec8c2191ca037d3f5b0662e21ae850debe8ea2",
+    "groupid": 1,
+    "partid": 1,
+    "total_payment_msat": 200000,
+    "attempt_msat": 100000,
+    "hops": [
+      {
+        "next_node": "035d2b1192dfba134e10e540875d366ebc8bc353d5aa766b80c090b39c3a5d885d",
+        "short_channel_id": "1x2x3",
+        "direction": 1,
+        "channel_in_msat": 100030,
+        "channel_out_msat": 100030
+      },
+      {
+        "next_node": "022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59",
+        "short_channel_id": "2x3x4",
+        "direction": 0,
```

Comment:
Is this the final destination as the direction is 0? And it charges 30 sats for receiving?

Reply:
No, the direction indicates which direction we want to traverse the edge: 0 goes from lexicographic smaller to larger (does not invert), and 1 goes from larger to smaller. It is slightly redundant since we have the node ID but I found it useful to cross-check against the gossip status, being able to tell which direction to look at